test/Intl/IntlPlatform.js

```diff
@@ -0,0 +1,102 @@
+ */
+function testSingleConstructor(test) {
+    return function (Ctor) {
+        const methods = [
+            "normalizeLanguageTag",
+            "isDTFLocaleAvailable",
+            "isNFLocaleAvailable",
+            "isCollatorLocaleAvailable",
+            "getDefaultLocale",
+            "compareString",
+            "formatNumber",
+            "cacheNumberFormat",
+            "formatDateTime",
+            "validateAndCanonicalizeTimeZone",
+            "getDefaultTimeZone",
+            "getPatternForSkeleton",
+            "getLocaleData",
+            "winglob",
+            "showPlatform",
+        ];
+        const cache = {};
+
+        for (const method of methods) {
+            cache[method] = Intl.platform[method];
+        }
+
+        test(Ctor);
+
+        for (const method of methods) {
+            Intl.platform[method] = cache[method];
+        }
+    }
+}
+
+/**
+ * Runs a safe, side effect-free test using every constructor
+ */
+function testEachConstructor(test) {
+    return function () {
+        for (const Ctor of [Intl.Collator, Intl.NumberFormat, Intl.DateTimeFormat]) {
+            testSingleConstructor(test)(Ctor);
+        }
+    }
+}
+
+/**
+ * Formats a message for tests that are run with testEachConstructor
+ */
+function message(Ctor, message) {
+    return `${message} (testing with ${Ctor.name})`;
+}
+
+testRunner.runTests([
+    {
+        name: "Changing the default locale",
+        body: testEachConstructor((Ctor) => {
+            platform.getDefaultLocale = () => "de-DE";
+            assert.areEqual("de-DE", new Ctor().resolvedOptions().locale, message(Ctor, "Default locale is respected with undefined language tag"));
+        })
+    },
+    {
+        name: "Limiting available locales",
+        body: testEachConstructor((Ctor) => {
+            const isXLocaleAvailableMap = {
+                Collator: "Collator",
+                NumberFormat: "NF",
+                DateTimeFormat: "DTF",
+            };
+
+            const mapped = isXLocaleAvailableMap[Ctor.name];
+            assert.isNotUndefined(mapped, `Invalid test setup: no mapped name available for ${Ctor.name}`);
+
+            platform[`is${mapped}LocaleAvailable`] = (langtag) => ["de", "de-DE", "en", "zh", "en-UK"].includes(langtag);
+            platform.getDefaultLocale = () => "en-UK";
+            assert.areEqual("en", new Ctor("en-US").resolvedOptions().locale, message(Ctor, "en-US should fall back to en"));
+            assert.areEqual("zh", new Ctor("zh-Hans").resolvedOptions().locale, message(Ctor, "zh-Hans should fall back to zh"));
+            assert.areEqual("de-DE", new Ctor("de-DE-gregory").resolvedOptions().locale, message(Ctor, "de-DE-gregory should fall back to de-DE"));
+            assert.areEqual("en-UK", new Ctor("sp").resolvedOptions().locale, message(Ctor, "An unknown language tag should fall back to the default"));
+        })
+    }
+], { verbose: false });

```

Comment:
Why not use the usual pattern of `-args summary -endargs` so that if the test goes wrong in the future it's easier to debug?

Reply:
I honestly don't think it makes the test any easier to debug, since problems are printed regardless. I have a change somewhere in some branch that actually modifies runTests() to take the default value of `verbose` as `WScript.Arguments.includes("summary")`...